src/libs/metadata_view.c

```diff
@@ -339,457 +350,667 @@ static gint _lib_metadata_sort_index(gconstpointer a, gconstpointer b)
 
-    char *exposure_str = dt_util_format_exposure(img->exif_exposure);
-    _metadata_update_value(md_exif_exposure, exposure_str, self);
-    g_free(exposure_str);
+      case md_exif_lens:
+        _metadata_update_value(md_exif_lens, img->exif_lens, self);
+        break;
 
-    if(isnan(img->exif_exposure_bias))
-    {
-      _metadata_update_value(md_exif_exposure_bias, NODATA_STRING, self);
-    }
-    else
-    {
-      snprintf(value, sizeof(value), _("%+.2f EV"), img->exif_exposure_bias);
-      _metadata_update_value(md_exif_exposure_bias, value, self);
-    }
+      case md_exif_aperture:
+        (void)g_snprintf(text, sizeof(text), "f/%.1f", (double)img->exif_aperture);
+        _metadata_update_value(md_exif_aperture, text, self);
+        break;
 
-    snprintf(value, sizeof(value), "%.0f mm", img->exif_focal_length);
-    _metadata_update_value(md_exif_focal_length, value, self);
+      case md_exif_exposure:
+      {
+        gchar *const str = dt_util_format_exposure(img->exif_exposure);
+        _metadata_update_value(md_exif_exposure, str, self);
+        g_free(str);
+      }
+      break;
 
-    if(isnan(img->exif_focus_distance) || fpclassify(img->exif_focus_distance) == FP_ZERO)
-    {
-      _metadata_update_value(md_exif_focus_distance, NODATA_STRING, self);
-    }
-    else
-    {
-      snprintf(value, sizeof(value), "%.2f m", img->exif_focus_distance);
-      _metadata_update_value(md_exif_focus_distance, value, self);
-    }
+      case md_exif_exposure_bias:
+        g_strlcpy(text, NODATA_STRING, sizeof(text));
+        if (!(isnan(img->exif_exposure_bias)))
+        {
+          (void)g_snprintf(text, sizeof(text), _("%+.2f EV"), (double)img->exif_exposure_bias);
+        }
+        _metadata_update_value(md_exif_exposure_bias, text, self);
+        break;
 
-    snprintf(value, sizeof(value), "%.0f", img->exif_iso);
-    _metadata_update_value(md_exif_iso, value, self);
+      case md_exif_focal_length:
+        (void)g_snprintf(text, sizeof(text), "%.0f mm", (double)img->exif_focal_length);
+        _metadata_update_value(md_exif_focal_length, text, self);
+        break;
 
-    struct tm tt_exif = { 0 };
-    if(sscanf(img->exif_datetime_taken, "%d:%d:%d %d:%d:%d", &tt_exif.tm_year, &tt_exif.tm_mon,
-      &tt_exif.tm_mday, &tt_exif.tm_hour, &tt_exif.tm_min, &tt_exif.tm_sec) == 6)
-    {
-      tt_exif.tm_year -= 1900;
-      tt_exif.tm_mon--;
-      tt_exif.tm_isdst = -1;
-      const time_t exif_timestamp = mktime(&tt_exif);
-      _metadata_update_timestamp(md_exif_datetime, &exif_timestamp, self);
-    }
-    else
-      _metadata_update_value(md_exif_datetime, img->exif_datetime_taken, self);
+      case md_exif_focus_distance:
+        (void)g_strlcpy(text, NODATA_STRING, sizeof(text));
+        if(!(isnan(img->exif_focus_distance) || (fpclassify(img->exif_focus_distance) == FP_ZERO) ))
+        {
+          (void)g_snprintf(text, sizeof(text), "%.2f m", (double)img->exif_focus_distance);
```

Comment:
As we are at it we probably want "%.2f m" to be translatable _("%.2f m") as in Russian IINM m is M.